spanner/tests/system/test_system.py

```diff
@@ -728,6 +728,20 @@ def _unit_of_work(transaction, test):
 
         return session, committed
 
+    def test_read_with_single_keys_index(self):
+        row_count = 10
+        columns = self.COLUMNS[1], self.COLUMNS[2]
+        session, committed = self._set_up_table(row_count)
+        self.to_delete.append(session)
+        expected = [[row[1], row[2]] for row in self._row_data(row_count)]
+        row = 5
+        rows = list(session.read(self.TABLE,
+                            columns,
+                            KeySet(keys=[[expected[row][0], expected[row][1]]]),
+                            index='name')
+                    )
```

Comment:
There is a lot (too much?) going on here, making it hard to read. Instead I recommend the following:

```python
XYZ = KeySet(keys=[[expected[row][0], expected[row][1]]])
read_iter = session.read(self.TABLE, columns, XYZ, index='name')
rows = list(read_iter)
```

You should replace `XYZ` with something that describes what the `KeySet` is doing